x-pack/plugins/endpoint/public/embeddables/resolver/models/process_event.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ProcessEvent } from '../types';
+
+export function isGraphableProcess(event: ProcessEvent) {
+  return eventType(event) === 'processCreated' || eventType(event) === 'processRan';
+}
+
+export function eventType(event: ProcessEvent) {
+  const {
+    data_buffer: { event_type_full: type, event_subtype_full: subType },
+  } = event;
+
+  if (type === 'process_event') {
+    if (subType === 'creation_event' || subType === 'fork_event' || subType === 'exec_event') {
+      return 'processCreated';
+    } else if (subType === 'already_running') {
+      return 'processRan';
+    } else if (subType === 'termination_event') {
+      return 'processTerminated';
+    } else {
+      return 'unknownProcessEvent';
+    }
+  } else if (type === 'alert_event') {
+    return 'processCausedAlert';
+  }
+  return 'unknownEvent';
+}
+
+export function uniquePidForProcess(event: ProcessEvent) {
+  return event.data_buffer.node_id;
+}
+
+export function uniqueParentPidForProcess(event: ProcessEvent) {
+  return event.data_buffer.source_id;
```

Comment:
I'd be very careful with nomenclature here, because this doesn't look like an "event" to me--"event"s are our raw data sources, for which we don't have a "source_id" field. This looks more like some sort of investigation or alert, something that originally came from the `hunt_data_schema`--if that's true then the applications for resolver out of the box are going to be very limited. Instead this should be working on the raw event data if possible. In which case:

1. `v0` (data coming from the SMP) will be looking at the field `endgame.unique_ppid` for process events
2. `v1` (data coming from Elastic-native sensor) will be looking at the field `endpoint.process.parent.entity_id` for events with `event.category == 'process'`

Reply:
Let's sync on all of this. But basically, this is using some random sample data that has pids, unique ppids, and process names. that's all that we need to test the layout logic. Once we get a BE, we'll get this stuff working against real data.